src/request.ts

```diff
@@ -77,27 +77,59 @@ export async function buildRequestUrl(
 }
 
 /**
- * The error type returned by `executeRequestError`.
+ * The error type returned by the SDK whenever it makes a network request
+ * (internally, this happens in `executeRequest`). It implements, so is
+ * compatible with, `AxiosError`.
  */
-export class ExecuteRequestError extends Error {
-  originalError: AxiosError;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class ExecuteRequestError<T = any, D = any> extends Error implements AxiosError {
   responseStatus: number | null;
   responseMessage: string | null;
 
+  // Properties required by `AxiosError`.
+
+  config: AxiosRequestConfig<D>;
+  code?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request?: any;
+  response?: AxiosResponse<T, D>;
+  isAxiosError: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  toJSON: () => object;
+
   /**
    * Creates an `ExecuteRequestError`.
    *
    * @param message - The error message.
-   * @param axiosError - The original axios error.
+   * @param axiosError - The original Axios error.
    * @param responseStatus - The response status, if found in the original error.
    * @param responseMessage - The response message, if found in the original error.
    */
   constructor(message: string, axiosError: AxiosError, responseStatus: number | null, responseMessage: string | null) {
+    // Include this check since ExecuteRequestError implements AxiosError, but
+    // we only expect original errors from Axios here.
+    if (axiosError instanceof ExecuteRequestError) {
+      throw new Error(
+        "Could not instantiate an `ExecuteRequestError` from an `ExecuteRequestError`, an original error from axios was expected"
+      );
+    }
```

Comment:
Is there a reason for us to introduce this limitation? I understand the "limit the scope" part of it but I'm not sure if there is any actual danger in allowing these errors to be daisy-chained.

Reply:
The danger that I saw is just that it's clearly a developer/logic bug to accept `ExecuteRequestError` here, even though it's possible due to the way it's typed. (Just went down a rabbithole trying to figure out a Typescript typing that includes `AxiosError` but excludes `ExecuteRequestError`. `Excludes<>` didn't work for some reason.) Anyways, updated the comment.